### PR TITLE
Upgrade GCLogParser to NetCoreAppToolCurrent

### PR DESCRIPTION
**Pull request by am11**
net472 -> net9

### CODE REVIEW

File: src/coreclr/tools/GCLogParser/parse-hb-log.cs [hunk 1]
```diff
@@ -495,59 +501,59 @@ static void PassZero(string strLog)
...
 
-        static void PrintAllocToAllPassOneFiles()
+        private static void PrintAllocToAllPassOneFiles()
         {
             string strAlloc = string.Format("{0,6}", nodeIndexToPrint);
             int procStart = procsPerNode * nodeIndexToPrint;
             int procEnd = procsPerNode * (nodeIndexToPrint + 1);
 
-            int currentGCAllocMBAllHeaps = 0;
+            // int currentGCAllocMBAllHeaps = 0;
```
Comment (mangod9) [inline src/coreclr/tools/GCLogParser/parse-hb-log.cs:556]: assuming these were unused locals, should the lines be deleted?

**Comment by mangod9** [comment]
Hi @am11, Was the motivation to add some functionality to this tool after moving to latest runtime?

File: src/coreclr/tools/GCLogParser/parse-hb-log.cs (hunk 1 above)
Comment (am11) [reply src/coreclr/tools/GCLogParser/parse-hb-log.cs:556]: Yup, I was getting warning (unnecessary assignments etc.).

File: src/coreclr/tools/GCLogParser/parse-hb-log.cs [hunk 2]
```diff
@@ -557,7 +563,7 @@ static void PrintAllocToAllPassOneFiles()
                 //    currentGCAllocMB);
             }
 
-            currentGCAllocMBAllHeaps = currentGCAllocMB;
+            // currentGCAllocMBAllHeaps = currentGCAllocMB;
```
Comment (am11) [inline src/coreclr/tools/GCLogParser/parse-hb-log.cs:566]:
```suggestion
```

**Comment by am11** [comment]
> Hi @am11, Was the motivation to add some functionality to this tool after moving to latest runtime?

It was motivated by https://github.com/dotnet/runtime/pull/101084, which in turn was motivated by https://github.com/dotnet/runtime/pull/101038 to cleanup unnecessary arch-specific builds (in this case `<PlatformTarget>x64</PlatformTarget>` etc.).

**Review by mangod9 — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**